src/citra_qt/util/vk_device_info.cpp

```diff
@@ -0,0 +1,23 @@
+// Copyright 2023 Citra Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#include "citra_qt/util/vk_device_info.h"
+#include "video_core/renderer_vulkan/vk_instance.h"
+
+std::vector<QString> GetVulkanPhysicalDevices() {
+    std::vector<QString> result;
+    try {
+        Vulkan::Instance instance{};
+        const auto physical_devices = instance.GetPhysicalDevices();
+
+        for (const vk::PhysicalDevice physical_device : physical_devices) {
+            const QString name = QString::fromLocal8Bit(physical_device.getProperties().deviceName);
```

Comment:
Isn't deviceName supposed to be `a null-terminated UTF-8 string`? Shouldn't `QString::fromUtf8` be used in that case?

Also a side note, from Qt5 -> Qt6 it seems like the (const char *str, int size) overload no longer has a default value for size, so this call will probably try to use the QByteArray overload instead, which will [convert](https://doc.qt.io/qt-6/qstring.html#fromLocal8Bit-2) the null bytes to unicode nulls, possibly explaining one behavior reported on discord with invalid characters being shown on the dropdown.

```suggestion
            const QString name = QString::fromUtf8(physical_device.getProperties().deviceName, -1);
```